nativelib/src/main/scala/scala/scalanative/native/pure.scala

```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package native
+
+/** Let optimizer assume that given method is pure.
+  * This means that repetitive calls to the same method
+  * with the same arguments are allowed to be eliminated.
+  * Additionally, if method result to given method is not used
+  * then the call can be elided altogether.
+  */
+final class pure extends scala.annotation.StaticAnnotation
```

Comment:
This one is really simple, just assume it's pure and eliminate module loads / function calls based on this assumption. If it is in fact side-effecting in observable ways, it's programmer's fault. Full-blown purity analysis probably wouldn't happen any time soon (and then this annotation will become mute) and we need some short-term solution to clean up of extraneous runtime module loads and runtime calls.